stress/src/logs.rs

```diff
@@ -37,14 +39,15 @@ pub struct MockLogProcessor {
 impl LogProcessor for MockLogProcessor {
     fn emit(&self, record: &mut opentelemetry_sdk::logs::LogRecord, scope: &InstrumentationScope) {
         let log_tuple = &[(record as &LogRecord, scope)];
-        let _ = futures_executor::block_on(self.exporter.export(LogBatch::new(log_tuple)));
+        let log_batch = LogBatch::new(log_tuple);
+        let _ = futures_executor::block_on(self.exporter.export(&log_batch));
```

Comment:
The overall flow looks complex..
emit gets a mutable ref to LogRecord. It is ref by design (and mut by design.)
processor, if need to batch it and send, can clone and keep the copy. When it calls the exporter, can it pass ownership the exporter? Wondering why we only pass a ref to the exporter? The processor has no reason to retain ownership anyway, so why not give it up, to the exporter?

Thinking out loud to see if this can be simplified without sacrificing perf..

Reply:
This is good point. We are passing `LogBatch` as reference. `LogBatch` is a lightweight wrapper around the slice of tuple of `LogRecord` and `InstrumentationScope`. Passing it by value won't change the ownership of this borrowed data (record and scope). We can change batch to be passed as value (though record and scope would still remain as reference). This can simplify the design (remove 'a need). However, reference just shows the intent better, and need to test the perf (theoretically it shouldn't affect).

ps - Need to dig the reason for passing record and scope as reference, I remember we discussed and agreed for it in earlier discussion. Here, we are only taking about passing batch as reference or value.

Reply:
>  The processor has no reason to retain ownership anyway, so why not give it up, to the exporter?

Probably it was not very clear from my above message. Just to clarify - this PR is not changing the ownership model. The exporter always gets the reference to the `LogRecord`, and it clones if the `LogRecord` needs to be used beyond the export function lifetime. Are we now discussing the existing design?

Reply:
> reference just shows the intent better,

Not so sure, since we are passing a custom structure `LogBatch`, which is defined to contains refs only.
Or we can pass slice of refs only, avoiding LogBatch structure entirely. This is better discussed separately.

> Are we now discussing the existing design?

This PR is changing existing design - previously owned LogBatch was passed, now we pass a &LogBatch only. It is not clear if that is required for using async native or it is unrelated to that.
If there is no need to change from passing LogBatch to &LogBatch to make async-native work, then lets revert that part.

Reply:
> If there is no need to change from passing LogBatch to &LogBatch to make async-native work, then lets revert that part.

This is reverted now.